### PR TITLE
Show route information in window title

### LINKED ISSUE #1605: Show route information in dashboard window title
**What would you like to be added**:

Add route metdata for the document title. 
* For Shoot instance pages show the shoot name
* For Project pages show the project name
* Show the name of the route like `<projectName> Members`

**Why is this needed**:

### PR DESCRIPTION
**Pull request by grolu**
**What this PR does / why we need it**:
Screenshot 2024-04-04 at 16 50 02

**Which issue(s) this PR fixes**:
Fixes #1605

**Special notes for your reviewer**:

**Release note**:

```feature user
Route information is now included in the browser window title
```

### CODE REVIEW

File: frontend/src/router/guards.js [hunk 1]
```diff
@@ -165,6 +165,18 @@ export function createGlobalBeforeGuards () {
...
+      titles.push(to.meta.title ?? to.name)
+      if (to.params.name) {
+        titles.push(to.params.name)
+      }
+      if (projectStore.projectName) {
+        titles.push(projectStore.projectName)
+      }
+      titles.push(process.env.VITE_APP_TITLE)
+      document.title = titles.join(' | ')
+    },
```
Comment (petersutter) [inline frontend/src/router/guards.js:179]: I think the title should match the breadcrumb

File: frontend/src/App.vue [hunk 2]
```diff
@@ -76,4 +78,32 @@ watch(visibility, (current, previous) => {
...
+  let titleItems = [pageTitle]
+  if (route.meta.namespaced !== false) {
+    const locationTitle = [projectName, routeParamName].filter(item => item !== undefined).join('/')
+    titleItems.push(locationTitle)
+  }
+  titleItems = titleItems.filter(item => item?.length)
+
+  if (titleItems.length) {
+    appTitle = `${titleItems.join(' • ')} | ${appTitle}`
+  }
```
Comment (holgerkoser) [inline frontend/src/App.vue:100]:
```suggestion
  const titleItems = []
  

  const pageTitle = route.meta.title ?? route.name
  if (pageTitle) {
    titleItems.push(pageTitle)
  }
  
  if (route.meta.namespaced !== false) {
    const projectName = projectStore.projectName
    const routeParamName = route.params.name
    if (routeParamName) {
      titleItems.push([projectName, routeParamName].join('/'))
    } else if (projectName) {
      titleItems.push(projectName)
    }
  }

  if (titleItems.length) {
    appTitle = `${titleItems.join(' • ')} | ${appTitle}`
  }
```

File: frontend/src/App.vue [hunk 3]
```diff
@@ -13,28 +13,36 @@ import {
...
 
 import { useConfigStore } from '@/store/config'
 import { useLoginStore } from '@/store/login'
 import { useLocalStorageStore } from '@/store/localStorage'
 import { useShootStore } from '@/store/shoot'
+import { useProjectStore } from '@/store/project'
 
 import { useCustomColors } from '@/composables/useCustomColors'
 
+import { last } from '@/lodash'
```
Comment (holgerkoser) [inline frontend/src/App.vue]: I think this is not needed anymore

**Review by holgerkoser — APPROVED** [review]
/lgtm

**Review by petersutter — APPROVED** [review]
/lgtm

**Pull request merged**